go/libraries/doltcore/sqle/schema_table.go

```diff
@@ -88,120 +178,45 @@ func getOrCreateDoltSchemasTable(ctx *sql.Context, db Database) (retTbl *Writabl
 		}
 	}
 
-	tbl, found, err := db.GetTableInsensitive(ctx, doltdb.SchemasTableName)
+	tbl, _, err := db.GetTableInsensitive(ctx, doltdb.SchemasTableName)
 	if err != nil {
 		return nil, err
 	}
 
-	if found {
-		schemasTable := tbl.(*WritableDoltTable)
-		// Old schemas table contains the `id` column or is missing an `extra` column.
-		if tbl.Schema().Contains(doltdb.SchemasTablesIdCol, doltdb.SchemasTableName) || !tbl.Schema().Contains(doltdb.SchemasTablesExtraCol, doltdb.SchemasTableName) {
-			return migrateOldSchemasTableToNew(ctx, db, schemasTable)
+	wrapper, ok := tbl.(*SchemaTable)
+	if !ok {
+		return nil, fmt.Errorf("expected a SchemaTable, but found %T", tbl)
+	}
+
+	if wrapper.backingTable != nil {
+		schemasTable := wrapper.backingTable
+		// Old schemas are missing the `extra` column. Very ancient. Provide error message and bail.
+		if !schemasTable.Schema().Contains(doltdb.SchemasTablesExtraCol, doltdb.SchemasTableName) {
+			return nil, fmt.Errorf("cannot migrate dolt_schemas table from v0.19.1 or earlier")
 		} else {
 			return schemasTable, nil
 		}
 	}
 
 	// Create new empty table
-	err = db.createDoltTable(ctx, doltdb.SchemasTableName, schemaName, root, schemaTableSchema)
-	if err != nil {
-		return nil, err
-	}
-	tbl, found, err = db.GetTableInsensitive(ctx, doltdb.SchemasTableName)
-	if err != nil {
-		return nil, err
-	}
-	if !found {
-		return nil, sql.ErrTableNotFound.New(doltdb.SchemasTableName)
-	}
-
-	return tbl.(*WritableDoltTable), nil
-}
-
-func migrateOldSchemasTableToNew(ctx *sql.Context, db Database, schemasTable *WritableDoltTable) (newTable *WritableDoltTable, rerr error) {
```

Comment:
I think we should try to preserve schema migration for `dolt_schemas`. The last schema change was for `sqlMode`, which was just shy of a year ago. That is for sure a long time to go without updating your Dolt version, but it is still after we launched 1.0, so seems like we should make an effort to keep those 1.0 versions working with new Dolt versions. 

That should just mean adding back the migrateOldSchemasTableToNew function and adding a test with a schema created without the sqlMode column.